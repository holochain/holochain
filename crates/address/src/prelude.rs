pub use address_types::Address;
pub use crate::*;
pub use crate::serial_address;
