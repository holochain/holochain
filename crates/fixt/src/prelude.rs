pub use crate::basic_test;
pub use crate::bool::BoolFixturator;
pub use crate::bytes::Bytes;
pub use crate::bytes::BytesFixturator;
pub use crate::bytes::BytesNotEmpty;
pub use crate::bytes::BytesNotEmptyFixturator;
pub use crate::bytes::SixtyFourBytesFixturator;
pub use crate::bytes::SixtyFourBytesVecFixturator;
pub use crate::bytes::ThirtySixBytesFixturator;
pub use crate::bytes::ThirtyTwoBytesFixturator;
pub use crate::curve;
pub use crate::enum_fixturator;
pub use crate::fixt;
pub use crate::fixturator;
pub use crate::get_fixt_curve;
pub use crate::get_fixt_index;
pub use crate::newtype_fixturator;
pub use crate::number::*;
pub use crate::serialized_bytes::SerializedBytesFixturator;
pub use crate::set_fixt_index;
pub use crate::string::CharFixturator;
pub use crate::string::StringFixturator;
pub use crate::unit::UnitFixturator;
pub use crate::wasm_io_fixturator;
pub use crate::Empty;
pub use crate::Fixturator;
pub use crate::Predictable;
pub use crate::Unpredictable;
pub use paste::paste;
pub use rand::prelude::*;
pub use strum::IntoEnumIterator;
pub use strum_macros;
