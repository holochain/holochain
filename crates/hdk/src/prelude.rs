pub use crate::app_entry;
pub use crate::capability::create_cap_claim;
pub use crate::capability::create_cap_grant;
pub use crate::capability::delete_cap_grant;
pub use crate::capability::generate_cap_secret;
pub use crate::capability::update_cap_grant;
pub use crate::chain::get_agent_activity;
pub use crate::chain::query;
pub use crate::ed25519::sign;
pub use crate::ed25519::sign_ephemeral;
pub use crate::ed25519::sign_ephemeral_raw;
pub use crate::ed25519::sign_raw;
pub use crate::ed25519::verify_signature;
pub use crate::ed25519::verify_signature_raw;
pub use crate::entry::create;
pub use crate::entry::create_entry;
pub use crate::entry::delete;
pub use crate::entry::delete_entry;
pub use crate::entry::get;
pub use crate::entry::get_details;
pub use crate::entry::hash_entry;
pub use crate::entry::must_get_entry;
pub use crate::entry::must_get_header;
pub use crate::entry::must_get_valid_element;
pub use crate::entry::update;
pub use crate::entry::update_entry;
pub use crate::entry::EntryDefRegistration;
pub use crate::entry_def;
pub use crate::entry_def_index;
pub use crate::entry_defs;
pub use crate::entry_type;
pub use crate::hash_path::anchor::anchor;
pub use crate::hash_path::anchor::get_anchor;
pub use crate::hash_path::anchor::list_anchor_addresses;
pub use crate::hash_path::anchor::list_anchor_tags;
pub use crate::hash_path::anchor::list_anchor_type_addresses;
pub use crate::hash_path::anchor::Anchor;
pub use crate::hash_path::path::Path;
pub use crate::hdk::*;
pub use crate::info::agent_info;
pub use crate::info::zome_info;
pub use crate::link::create_link;
pub use crate::link::delete_link;
pub use crate::link::get_link_details;
pub use crate::link::get_links;
pub use crate::map_extern;
pub use crate::map_extern::ExternResult;
pub use crate::p2p::call;
pub use crate::p2p::call_remote;
pub use crate::p2p::emit_signal;
pub use crate::p2p::remote_signal;
pub use crate::random::*;
pub use crate::register_entry;
pub use crate::time::schedule;
pub use crate::time::sleep;
pub use crate::time::sys_time;
pub use crate::time::*;
pub use crate::x_salsa20_poly1305::create_x25519_keypair;
pub use crate::x_salsa20_poly1305::x_25519_x_salsa20_poly1305_decrypt;
pub use crate::x_salsa20_poly1305::x_25519_x_salsa20_poly1305_encrypt;
pub use crate::x_salsa20_poly1305::x_salsa20_poly1305_decrypt;
pub use crate::x_salsa20_poly1305::x_salsa20_poly1305_encrypt;
pub use hdk_derive::hdk_entry;
pub use hdk_derive::hdk_extern;
pub use holo_hash;
pub use holo_hash::AgentPubKey;
pub use holo_hash::AnyDhtHash;
pub use holo_hash::EntryHash;
pub use holo_hash::EntryHashes;
pub use holo_hash::HasHash;
pub use holo_hash::HeaderHash;
pub use holo_hash::HoloHash;
pub use holo_hash::HoloHashed;
pub use holochain_wasmer_guest::*;
pub use holochain_zome_types;
pub use holochain_zome_types::prelude::*;
pub use std::collections::BTreeSet;
pub use std::collections::HashSet;
pub use std::convert::TryFrom;
pub use tracing;
pub use tracing::{debug, error, info, instrument, trace, warn};
pub use tracing_subscriber;

#[cfg(feature = "mock")]
pub use mockall;

// This needs to be called at least once _somewhere_ and is idempotent.
#[macro_export]
macro_rules! holochain_externs {
    () => {
        holochain_wasmer_guest::host_externs!(
            __trace,
            __hash_entry,
            __unreachable,
            __verify_signature,
            __sign,
            __sign_ephemeral,
            __zome_info,
            __call_info,
            __dna_info,
            __app_info,
            __random_bytes,
            __sys_time,
            __agent_info,
            __capability_claims,
            __capability_grants,
            __capability_info,
            __get,
            __get_details,
            __get_links,
            __get_link_details,
            __get_agent_activity,
            __must_get_entry,
            __must_get_valid_element,
            __must_get_header,
            __query,
            __call_remote,
            __call,
            __create,
            __emit_signal,
            __remote_signal,
            __create_link,
            __delete_link,
            __update,
            __delete,
            __schedule,
            __sleep,
            __x_salsa20_poly1305_encrypt,
            __x_salsa20_poly1305_decrypt,
            __x_25519_x_salsa20_poly1305_encrypt,
            __x_25519_x_salsa20_poly1305_decrypt,
            __create_x25519_keypair
        );
    };
}

#[cfg(not(feature = "mock"))]
holochain_externs!();
