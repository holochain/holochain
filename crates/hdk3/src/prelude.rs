pub use crate::capability::create_cap_claim::create_cap_claim;
pub use crate::capability::create_cap_grant::create_cap_grant;
pub use crate::capability::delete_cap_grant::delete_cap_grant;
pub use crate::capability::generate_cap_secret::generate_cap_secret;
pub use crate::capability::update_cap_grant::update_cap_grant;
pub use crate::debug;
pub use crate::entry::create_entry::create_entry;
pub use crate::entry::delete_entry::delete_entry;
pub use crate::entry::hash_entry::hash_entry;
pub use crate::entry::update_entry::update_entry;
pub use crate::entry_def;
pub use crate::entry_defs;
pub use crate::hash_path::anchor::anchor;
pub use crate::hash_path::anchor::get_anchor;
pub use crate::hash_path::anchor::list_anchor_addresses;
pub use crate::hash_path::anchor::list_anchor_tags;
pub use crate::hash_path::anchor::list_anchor_type_addresses;
pub use crate::hash_path::anchor::Anchor;
pub use crate::hash_path::path::Path;
pub use crate::host_fn::agent_info::agent_info;
pub use crate::host_fn::call::call;
pub use crate::host_fn::call_remote::call_remote;
pub use crate::host_fn::create::create;
pub use crate::host_fn::create_link::create_link;
pub use crate::host_fn::delete::delete;
pub use crate::host_fn::delete_link::delete_link;
pub use crate::host_fn::emit_signal::emit_signal;
pub use crate::host_fn::get::get;
pub use crate::host_fn::get_agent_activity::get_agent_activity;
pub use crate::host_fn::get_details::get_details;
pub use crate::host_fn::get_link_details::get_link_details;
pub use crate::host_fn::get_links::get_links;
pub use crate::host_fn::query::query;
pub use crate::host_fn::random_bytes::random_bytes;
pub use crate::host_fn::random_bytes::*;
pub use crate::host_fn::remote_signal::remote_signal;
pub use crate::host_fn::sign::sign;
pub use crate::host_fn::sys_time::sys_time;
pub use crate::host_fn::update::update;
pub use crate::host_fn::verify_signature::verify_signature;
pub use crate::host_fn::zome_info::zome_info;
pub use crate::map_extern;
pub use crate::map_extern::ExternResult;
pub use crate::x_salsa20_poly1305::create_x25519_keypair::create_x25519_keypair;
pub use crate::x_salsa20_poly1305::x_25519_x_salsa20_poly1305_decrypt::x_25519_x_salsa20_poly1305_decrypt;
pub use crate::x_salsa20_poly1305::x_25519_x_salsa20_poly1305_encrypt::x_25519_x_salsa20_poly1305_encrypt;
pub use crate::x_salsa20_poly1305::x_salsa20_poly1305_decrypt::x_salsa20_poly1305_decrypt;
pub use crate::x_salsa20_poly1305::x_salsa20_poly1305_encrypt::x_salsa20_poly1305_encrypt;
pub use hdk3_derive::hdk_entry;
pub use hdk3_derive::hdk_extern;
pub use holo_hash::AgentPubKey;
pub use holo_hash::AnyDhtHash;
pub use holo_hash::EntryHash;
pub use holo_hash::EntryHashes;
pub use holo_hash::HasHash;
pub use holo_hash::HeaderHash;
pub use holo_hash::HoloHash;
pub use holochain_wasmer_guest::*;
pub use holochain_zome_types;
pub use holochain_zome_types::prelude::*;
pub use std::collections::HashSet;
pub use std::collections::BTreeSet;
pub use std::collections::BTreeMap;
pub use std::convert::TryFrom;

// This needs to be called at least once _somewhere_ and is idempotent.
#[macro_export]
macro_rules! holochain_externs {
    () => {
        holochain_wasmer_guest::memory_externs!();
        holochain_wasmer_guest::host_externs!(
            __debug,
            __hash_entry,
            __unreachable,
            __verify_signature,
            __sign,
            __decrypt,
            __encrypt,
            __zome_info,
            __property,
            __random_bytes,
            __show_env,
            __sys_time,
            __agent_info,
            __capability_claims,
            __capability_grants,
            __capability_info,
            __get,
            __get_details,
            __get_links,
            __get_link_details,
            __get_agent_activity,
            __query,
            __call_remote,
            __call,
            __create,
            __emit_signal,
            __remote_signal,
            __create_link,
            __delete_link,
            __update,
            __delete,
            __schedule,
            __x_salsa20_poly1305_encrypt,
            __x_salsa20_poly1305_decrypt,
            __x_25519_x_salsa20_poly1305_encrypt,
            __x_25519_x_salsa20_poly1305_decrypt,
            __create_x25519_keypair
        );
    };
}

holochain_externs!();
