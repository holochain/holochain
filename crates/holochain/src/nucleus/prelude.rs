pub use super::dna::zome::Zome;
