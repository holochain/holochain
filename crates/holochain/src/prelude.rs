pub use crate::nucleus::prelude::*;
pub use holochain_types::prelude::*;
