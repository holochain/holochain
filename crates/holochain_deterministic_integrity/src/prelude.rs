pub use crate::app_entry;
pub use crate::ed25519::verify_signature;
pub use crate::ed25519::verify_signature_raw;
pub use crate::entry::must_get_entry;
pub use crate::entry::must_get_header;
pub use crate::entry::must_get_valid_element;
pub use crate::entry::EntryDefRegistration;
pub use crate::entry_def;
pub use crate::entry_def_index;
pub use crate::entry_defs;
pub use crate::entry_type;
pub use crate::hash::*;
pub use crate::hdi::*;
pub use crate::info::dna_info;
pub use crate::info::zome_info;
pub use crate::link::HdkLinkType;
pub use crate::map_extern;
pub use crate::map_extern::ExternResult;
pub use crate::map_extern_infallible;
pub use crate::register_entry;
pub use crate::x_salsa20_poly1305::x_25519_x_salsa20_poly1305_decrypt;
pub use crate::x_salsa20_poly1305::x_salsa20_poly1305_decrypt;
pub use hdk_derive::hdk_entry;
pub use hdk_derive::hdk_extern;
pub use holo_hash;
pub use holo_hash::AgentPubKey;
pub use holo_hash::AnyDhtHash;
pub use holo_hash::EntryHash;
pub use holo_hash::EntryHashes;
pub use holo_hash::HasHash;
pub use holo_hash::HeaderHash;
pub use holo_hash::HoloHash;
pub use holo_hash::HoloHashed;
pub use holochain_integrity_types;
pub use holochain_integrity_types::prelude::*;
pub use holochain_wasmer_guest::*;
pub use std::collections::BTreeSet;
pub use std::collections::HashSet;
pub use std::convert::TryFrom;
#[cfg(feature = "trace")]
pub use tracing;
#[cfg(feature = "trace")]
pub use tracing::{debug, error, info, instrument, trace, warn};

#[cfg(not(feature = "trace"))]
/// Needed as a noop for map_extern! when trace is off.
pub use crate::error;

#[cfg(not(feature = "trace"))]
#[macro_export]
/// Needed as a noop for map_extern! when trace is off.
macro_rules! error {
    ($($field:tt)*) => {};
}

#[cfg(feature = "mock")]
pub use mockall;

// This needs to be called at least once _somewhere_ and is idempotent.
#[macro_export]
macro_rules! holochain_externs {
    () => {
        holochain_wasmer_guest::host_externs!(
            __trace,
            __hash,
            __unreachable,
            __verify_signature,
            __zome_info,
            __dna_info,
            __must_get_entry,
            __must_get_valid_element,
            __must_get_header,
            __x_salsa20_poly1305_decrypt,
            __x_25519_x_salsa20_poly1305_decrypt
        );
    };
}

#[cfg(not(feature = "mock"))]
holochain_externs!();
