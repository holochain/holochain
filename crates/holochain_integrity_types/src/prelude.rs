//! Common types

pub use crate::action::conversions::*;
pub use crate::action::*;
pub use crate::capability::*;
pub use crate::countersigning::*;
pub use crate::entry::*;
pub use crate::entry_def::*;
pub use crate::genesis::*;
pub use crate::hash::*;
pub use crate::info::*;
pub use crate::link::*;
pub use crate::op::*;
pub use crate::rate_limit::*;
pub use crate::record::*;
pub use crate::signature::*;
pub use crate::timestamp::*;
pub use crate::trace::*;
pub use crate::validate::*;
pub use crate::x_salsa20_poly1305::data::*;
pub use crate::x_salsa20_poly1305::encrypted_data::*;
pub use crate::x_salsa20_poly1305::key_ref::*;
pub use crate::x_salsa20_poly1305::x25519::*;
pub use crate::x_salsa20_poly1305::*;
pub use crate::zome::*;
pub use crate::zome_io::ExternIO;
pub use crate::zome_io::*;
pub use crate::*;
