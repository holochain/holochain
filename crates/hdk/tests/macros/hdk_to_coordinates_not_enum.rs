
use hdk::prelude::holochain_deterministic_integrity::prelude::*;

#[hdk_to_coordinates]
struct Foo;

fn main() {}
