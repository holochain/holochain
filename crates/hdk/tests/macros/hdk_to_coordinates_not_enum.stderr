error: The `hdk_to_coordinates` macro can only be used on enums.

         = help: Make this an enum.

 --> tests/macros/hdk_to_coordinates_not_enum.rs:5:1
  |
5 | struct Foo;
  | ^^^^^^^^^^^
